src/metkit/codes/api/CodesTypes.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * (C) Copyright 2025- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <variant>
+#include <vector>
+
+
+namespace metkit::codes {
+
+struct CodesHandlePtr;
+
+class CodesException : public std::exception {
```

Comment:
Needs to inherit from `eckit::Exception`

Reply:
Done.